crates/burn-train/src/metric/confusion_matrix.rs

```diff
@@ -0,0 +1,161 @@
+use super::{ClassificationAverage, ClassificationInput};
+use burn_core::prelude::{Backend, Int, Tensor};
+use std::fmt::{self, Debug};
+
+#[derive(Clone)]
+pub struct ConfusionMatrix<B: Backend> {
+    confusion_matrix_map: Tensor<B, 2, Int>,
+    classification_average: ClassificationAverage,
+}
+
+impl<B: Backend> Debug for ConfusionMatrix<B> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let to_vec = |tensor_data: Tensor<B, 1>| {
+            tensor_data
+                .to_data()
+                .to_vec::<f32>()
+                .expect("A vector representation of the input Tensor is expected")
+        };
+        let ratio_of_support_vec =
+            |metric: Tensor<B, 1>| to_vec(self.clone().ratio_of_support(metric));
+        f.debug_struct("ConfusionMatrix")
+            .field("tp", &ratio_of_support_vec(self.clone().true_positive()))
+            .field("fp", &ratio_of_support_vec(self.clone().false_positive()))
+            .field("tn", &ratio_of_support_vec(self.clone().true_negative()))
+            .field("fn", &ratio_of_support_vec(self.clone().false_negative()))
+            .field("support", &to_vec(self.clone().support()))
+            .finish()
+    }
+}
+
+impl<B: Backend> ConfusionMatrix<B> {
+    pub fn new(
+        input: &ClassificationInput<B>,
+        threshold: f64,
+        classification_average: ClassificationAverage,
+    ) -> Self {
+        let thresholded_predictions = input.predictions.clone().greater_elem(threshold);
```

Comment:
This assumes the input has been transformed to the `[0, 1]` range (e.g., post sigmoid activation for the binary case).

We should either add the option to apply a sigmoid before, or/and document the usage to state that outputs are expected to be in that range.

Reply:
I added usage documentation.